File: OrchidCore/src/main/java/com/eden/orchid/impl/server/admin/lists/OptionsList.java
package com.eden.orchid.impl.server.admin.lists;

import com.eden.orchid.api.options.OrchidFlag;
import com.eden.orchid.api.options.OrchidFlags;
import com.eden.orchid.api.server.admin.AdminList;

import javax.inject.Inject;
import java.util.Collection;

public final class OptionsList implements AdminList<OrchidFlag> {

    private final Collection<OrchidFlag> list;

    @Inject
    public OptionsList() {
        list = OrchidFlags.getInstance().getFlags();
    }

    @Override
    public String getKey() {
        return "options";
    }

    @Override
    public Collection<OrchidFlag> getItems() {
        return list;
    }

    @Override
    public String getItemId(OrchidFlag item) {
        return item.getClass().getSimpleName();
    }

    @Override
    public OrchidFlag getItem(String id) {
        for(OrchidFlag item : getItems()) {
            if(id.equals(item.getClass().getSimpleName())) {
                return item;
            }
        }
        return null;
    }
}

File: plugins/OrchidPluginDocs/src/main/kotlin/com/eden/orchid/plugindocs/lists/OptionsList.kt
package com.eden.orchid.plugindocs.lists

import com.eden.orchid.api.options.OrchidFlag
import com.eden.orchid.api.options.OrchidFlags
import com.eden.orchid.api.server.admin.AdminList

import javax.inject.Inject

@JvmSuppressWildcards
class OptionsList @Inject
constructor() : AdminList {

    private val list: Collection<OrchidFlag>

    init {
        list = OrchidFlags.getInstance().flags
    }

    override fun getKey(): String {
        return "options"
    }

    override fun getItems(): Collection<Any> {
        return list
    }

    override fun getItemId(item: Any): String {
        return item.javaClass.simpleName
    }

    override fun getItem(id: String): Any? {
        for (item in items) {
            if (id == item.javaClass.simpleName) {
                return item
            }
        }
        return null
    }
}

